tests/suites/test_suite_x509parse.function

```diff
@@ -1549,82 +1549,97 @@ exit:
+        TEST_EQUAL(crt.subject_key_id.len, 0);
     }
 
 exit:
     mbedtls_x509_crt_free(&crt);
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_X509_CRT_PARSE_C */
-void x509_crt_parse_authoritykeyid(data_t *buf,
-                                   int keyIdLength_arg,
+/* BEGIN_CASE depends_on:MBEDTLS_X509_CRT_PARSE_C:MBEDTLS_FS_IO */
+void x509_crt_parse_authoritykeyid(char *file,
+                                   data_t *keyId,
                                    char *authorityKeyId_issuer,
-                                   int serialLength_arg,
+                                   data_t *serial,
                                    int ref_ret)
 {
     mbedtls_x509_crt crt;
     int bufferCounter = 0;
     size_t issuerCounter = 0;
-    size_t keyIdLength = keyIdLength_arg;
-    size_t serialLength = serialLength_arg;
     unsigned int result = 0;
     mbedtls_x509_subject_alternative_name san;
     mbedtls_x509_name *pname = NULL;
 
     mbedtls_x509_crt_init(&crt);
 
-    TEST_ASSERT(mbedtls_x509_crt_parse_der(&crt, buf->x, buf->len) == ref_ret);
+    TEST_EQUAL(mbedtls_x509_crt_parse_file(&crt, file), ref_ret);
 
     if (ref_ret == 0) {
         /* KeyId test */
-        TEST_ASSERT(crt.authority_key_id.keyIdentifier.tag == MBEDTLS_ASN1_OCTET_STRING);
-        TEST_ASSERT(crt.authority_key_id.keyIdentifier.len == keyIdLength);
+        if (keyId->len > 0) {
+            TEST_EQUAL(crt.authority_key_id.keyIdentifier.tag, MBEDTLS_ASN1_OCTET_STRING);
+            TEST_EQUAL(memcmp(crt.authority_key_id.keyIdentifier.p, keyId->x, keyId->len), 0);
+            TEST_EQUAL(crt.authority_key_id.keyIdentifier.len, keyId->len);
+        } else {
+            TEST_EQUAL(crt.authority_key_id.keyIdentifier.tag, 0);
+            TEST_EQUAL(crt.authority_key_id.keyIdentifier.len, 0);
+        }
+
 
         /* Issuer test */
-        mbedtls_x509_sequence *issuerPtr = &crt.authority_key_id.authorityCertIssuer;
+        if (strlen(authorityKeyId_issuer) > 0) {
+            mbedtls_x509_sequence *issuerPtr = &crt.authority_key_id.authorityCertIssuer;
 
-        TEST_ASSERT(mbedtls_x509_parse_subject_alt_name(&issuerPtr->buf, &san) == 0);
+            TEST_EQUAL(mbedtls_x509_parse_subject_alt_name(&issuerPtr->buf, &san), 0);
 
-        pname = &san.san.directory_name;
+            pname = &san.san.directory_name;
 
-        while (pname != NULL) {
-            for (issuerCounter = 0; issuerCounter < pname->val.len; issuerCounter++) {
-                result |=
-                    (authorityKeyId_issuer[bufferCounter++] != pname->val.p[issuerCounter]);
+            while (pname != NULL) {
+                for (issuerCounter = 0; issuerCounter < pname->val.len; issuerCounter++) {
+                    result |=
+                        (authorityKeyId_issuer[bufferCounter++] != pname->val.p[issuerCounter]);
+                }
+                bufferCounter++; /* Skipping the slash */
+                pname = pname->next;
             }
```

Comment:
Not a blocker for this PR, but I'd prefer using standard functions for comparison of directory names.

I think we can either:
* Use `mbedtls_x509_string_to_names()` to convert the test string to a DN, then use `x509_name_cmp()` to compare to the name in the cert. This requires exporting `x509_name_cmp()` publicly.
* Use `mbedtls_x509_dn_gets()` to convert the name in the cert to a string and `strcmp()` with the test name.

But I'd be happy for this change to be in a followup PR.